libs/Modal/create/show/index.tsx

```diff
@@ -0,0 +1,170 @@
+import ReactDOM from "react-dom";
+import { createRoot, hydrateRoot } from "react-dom/client";
+import { DEFAULT_PORTAL_ID_PREFIX } from "../../constants";
+
+export const DEFAULT_ADAPTERS: FunctionComponent<{ children: ReactNode }> = (
+  props
+) => <>{props.children}</>;
+
+type Status = ReturnType<typeof createStatus[keyof typeof createStatus]>;
+
+const createStatus = {
+  INITIAL: () => ({
+    isLoading: false,
+    isSuccess: false,
+    isError: false,
+    error: null,
+    data: null,
+  }),
+  PENDING: () => ({
+    isLoading: true,
+    isSuccess: false,
+    isError: false,
+    error: null,
+    data: null,
+  }),
+  SUCCESS: (data: unknown) => ({
+    isLoading: false,
+    isSuccess: true,
+    isError: false,
+    error: null,
+    data,
+  }),
+  FAILURE: (error: unknown) => ({
+    isLoading: false,
+    isSuccess: false,
+    isError: true,
+    error,
+    data: null,
+  }),
+};
+
+const EVENT = {
+  // 추가 가능
+  onClick: "onClick",
+  onTouchStart: "onTouchStart",
+  onMouseDown: "onMouseDown",
+  onMouseEnter: "onMouseEnter",
+} as const;
+
+type ValueOfEVENT = typeof EVENT[keyof typeof EVENT];
+
+type Callback = () => void | Promise<void>;
+
+interface ShowOptions {
+  portalId?: string;
+  adapters: ComponentType<{ children: ReactNode }>;
+}
+
+const show = (
+  createModal: FunctionComponent<{
+    createClose: <T extends ElementType = "button", D = any>(
+      callback?: () => D | Promise<D>,
+      options?: {
+        onStart?: () => void | Promise<void>;
+        onSuccess?: (data: D) => void | Promise<void>;
+        onFailure?: (e: unknown) => void | Promise<void>;
+        onClosed?: () => void | Promise<void>;
+      }
+    ) => (
+      eventName?: ValueOfEVENT,
+      injectAsyncStatus?: (
+        status:
+          | ReturnType<typeof createStatus.INITIAL>
+          | ReturnType<typeof createStatus.PENDING>
+          | ReturnType<typeof createStatus.SUCCESS>
+          | ReturnType<typeof createStatus.FAILURE>
+      ) => Partial<ComponentProps<T>>
+    ) => Partial<{
+      [key in keyof typeof EVENT]: Callback;
+    }>;
+  }>,
+  options: ShowOptions = {
+    portalId: DEFAULT_PORTAL_ID_PREFIX,
+    adapters: DEFAULT_ADAPTERS,
+  }
+) => {
+  const {
+    portalId = DEFAULT_PORTAL_ID_PREFIX,
+    adapters: Adapters = DEFAULT_ADAPTERS,
+  } = options;
+
+  const id =
+    typeof portalId === "string" && portalId !== DEFAULT_PORTAL_ID_PREFIX
+      ? portalId
+      : `${DEFAULT_PORTAL_ID_PREFIX}-${new Date().getTime()}`;
+
+  let portalEl = document.getElementById(id);
+
+  if (!portalEl) {
+    portalEl = document.createElement("div");
+    portalEl.id = id;
+    portalEl.style.left = "0";
+    portalEl.style.right = "0";
+    portalEl.style.top = "0";
+    portalEl.style.bottom = "0";
+    portalEl.style.position = "fixed";
+    portalEl.style.display = "flex";
+    portalEl.style.flexDirection = "column";
+    portalEl.style.justifyContent = "center";
+    portalEl.style.alignItems = "center";
```

Comment:
(minor) 아래와 같이 여러 번의 style 선언을 하나로 줄이는 건 어떨까? 근데 이건 개인차라 보고 별로면 패스해줘!

```suggestion
    portalEl.style.cssText = `
      left: 0;
      right: 0;
      top: 0;
      bottom: 0;
      position: fixed;
      display: flex;
      flex-direction: column;
      justify-content: center;
      alignItems: center;
    `;
```

Reply:
오 cssText라는 api도 있었구나 훨씬 좋네용 바로커밋 넣어버려~

https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration/cssText